File: core/src/main/java/com/eloja/core/entity/Endereco.java
package com.eloja.core.entity;

import jakarta.persistence.*;
import lombok.Data;

@Data
@Entity
public class Endereco {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    Integer id;

    @Column(nullable = false)
    private String cep;

    @Column(nullable = false)
    private String logradouro;

    @Column(nullable = false)
    private String numero;

    private String complemento;

    @Column(nullable = false)
    private String bairro;

    @Column(nullable = false, length = 7)
    private String ibge;

    @Column(nullable = false, length = 2)
    private String uf;
}

File: frontend-react/src/models/Endereco.ts
class Endereco {
  id?: number;
  cep: string;
  logradouro: string;
  numero: string;
  complemento?: string;
  bairro: string;
  ibge: string;
  uf: string;

  constructor(
    cep: string,
    logradouro: string,
    numero: string,
    complemento: string,
    bairro: string,
    ibge: string,
    uf: string,
    id?: number,
  ) {
    this.id = id;
    this.cep = cep;
    this.logradouro = logradouro;
    this.numero = numero;
    this.complemento = complemento;
    this.bairro = bairro;
    this.ibge = ibge;
    this.uf = uf;
  }
}

export default Endereco;
